src/charm.py

```diff
@@ -121,92 +126,116 @@ def _get_database_relation_info(self) -> dict:
             "endpoints": relation_data["endpoints"],
         }
 
-    def _update_layer(self) -> None:
-        """Updates the Pebble configuration layer and config if changed."""
-        self.unit.status = MaintenanceStatus("Applying pebble layer")
+    def _run_sql_migration(self, set_timeout) -> None:
+        """Runs a command to create SQL schemas and apply migration plans."""
+        process = self._container.exec(
+            ["hydra", "migrate", "sql", "-e", "--config", self._hydra_config_path, "--yes"],
+            timeout=20.0 if set_timeout else None,
+        )
+        try:
+            stdout, _ = process.wait_output()
+            logger.info(f"Executing automigration: {stdout}")
+        except ExecError as err:
+            logger.error(f"Exited with code {err.exit_code}. Stderr: {err.stderr}")
+            self.unit.status = BlockedStatus("Database migration job failed")
```

Comment:
Do we really want the outcome to be Blocked when the user runs the action manually? It should probably only be event.fail() in those cases. Error handling should probably be handled in the event methods and the generic method here should not set status.

Reply:
I see your point, but if the user tries to run a migration and that fails, doesn't it mean that the application is blocked? Assuming it's not a juju related error, if the migration fails to run it probably means that (all or some) hydra queries might fail.

Perhaps a better approach would be to set a flag with the latest migration that has been run in the peer relation (maybe use the app version?) and then use that in the `update_status` hook to set the app's status. 
e.g. `self.app.status = (migration_version == app_version)`

Reply:
With recent changes, if the migration run by charm fails (e.g. due to timeout or missing config file), the status will be set to blocked and suggest in logs that the action should be run instead. If this one fails as well, the event will fail but no status will be set. Let me know if that looks better now.

Reply:
If the user runs the action too early for example, when the workload is not yet ready (let's say it's in maintenance), here we wouldn't want the action to place the charm in a blocked state but to simply fail and return a message like "can't complete the action because the workload is not running yet".